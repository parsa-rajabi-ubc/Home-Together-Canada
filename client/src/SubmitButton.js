import React from 'react';

export function SubmitButton(props){
    return(
        <label>
            {props.label}
            <input type="submit" value="Submit" />
        </label>
    );
}