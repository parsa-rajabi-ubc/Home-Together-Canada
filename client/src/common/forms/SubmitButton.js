/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.05
 *
 * @Description: Submit button input Form Component
 *
 */

import React from 'react';
import propTypes from "prop-types";

function SubmitButton(props){
    const { className, label, inputValue, onClick } = props;
    return(
        <label>
            {label}
            <input className={className} type="submit" value={inputValue} onClick={onClick}/>
        </label>
    );
}

SubmitButton.propTypes = {
    onClick: propTypes.func.isRequired,
    label: propTypes.string,
    className: propTypes.string,
    inputValue: propTypes.string
};

export default SubmitButton;
