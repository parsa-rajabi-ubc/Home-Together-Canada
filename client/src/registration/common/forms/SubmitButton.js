import React from 'react';
import propTypes from "prop-types";

function SubmitButton(props){
    return(
        <label>
            {props.label}
            <input type="submit" value="Submit" />
        </label>
    );
}

SubmitButton.PropTypes = {
    label: propTypes.string.isRequired
}

export default SubmitButton