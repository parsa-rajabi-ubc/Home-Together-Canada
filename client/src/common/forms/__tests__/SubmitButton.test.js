import React from 'react';
import renderer from  'react-test-renderer'
import SubmitButton from "../SubmitButton";
/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.13
 *
 * @Description: Submit Button Component Snapshot test
 *
 */
describe('SubmitButton', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            //when
            const component = renderer.create(<SubmitButton label="testLabelString1" />).toJSON();
            //then
            expect(component).toMatchSnapshot();
        });
    })
})