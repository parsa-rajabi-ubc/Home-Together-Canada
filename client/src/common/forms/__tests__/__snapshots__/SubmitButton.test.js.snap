// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton Snapshot test should render correctly regardless of properties 1`] = `
<label>
  testLabelString1
  <input
    className="testClassNameString"
    disabled={false}
    onClick={[MockFunction]}
    type="submit"
    value="Submit"
  />
</label>
`;
